x-pack/plugin/data-frame/qa/single-node-tests/src/test/java/org/elasticsearch/xpack/dataframe/integration/DataFrameTaskFailedStateIT.java

```diff
@@ -0,0 +1,103 @@
+        assertThat(ex.getResponse().getStatusLine().getStatusCode(), equalTo(RestStatus.CONFLICT.getStatus()));
+        assertThat(XContentMapValues.extractValue("error.reason", entityAsMap(ex.getResponse())),
+            equalTo("Unable to stop data frame transform [failure_pivot_1] as it is in a failed state with reason: [" +
+                "no such index [reviews]]. Use force stop to stop the data frame transform."));
+
+        ex = expectThrows(ResponseException.class, () -> startDataframeTransform(transformId, false));
+        assertThat(ex.getResponse().getStatusLine().getStatusCode(), equalTo(RestStatus.CONFLICT.getStatus()));
+        assertThat(XContentMapValues.extractValue("error.reason", entityAsMap(ex.getResponse())),
+            equalTo("Unable to start data frame transform [failure_pivot_1] as it is in a failed state with failure: [" +
+                "no such index [reviews]]. Use force start to restart data frame transform once error is resolved."));
+
+        // Correct the failure by creating the reviews index again
+        createReviewsIndex();
+        // Force start the data frame to indicate failure correction
+        startDataframeTransform(transformId, true);
+        // Wait for data to be indexed appropriately and refresh for search
+        waitForDataFrameGeneration(transformId);
+        refreshIndex(dataFrameIndex);
+
+        // Verify that we have started and that our reason is cleared
+        fullState = getDataFrameState(transformId);
+        assertThat(XContentMapValues.extractValue("state.reason", fullState), is(nullValue()));
+        assertThat(XContentMapValues.extractValue("state.state", fullState), equalTo("started"));
+        assertThat(XContentMapValues.extractValue("state.transform_state", fullState), equalTo("started"));
+        assertThat(XContentMapValues.extractValue("stats.search_failures", fullState), equalTo(1));
+
+        // get and check some users to verify we restarted
+        assertOnePivotValue(dataFrameIndex + "/_search?q=reviewer:user_0", 3.776978417);
+        assertOnePivotValue(dataFrameIndex + "/_search?q=reviewer:user_5", 3.72);
+        assertOnePivotValue(dataFrameIndex + "/_search?q=reviewer:user_11", 3.846153846);
+        assertOnePivotValue(dataFrameIndex + "/_search?q=reviewer:user_20", 3.769230769);
+        assertOnePivotValue(dataFrameIndex + "/_search?q=reviewer:user_26", 3.918918918);
+
+
+        stopDataFrameTransform(transformId, true);
+        deleteDataFrameTransform(transformId);
+    }
+
+    private void awaitState(String transformId, DataFrameTransformTaskState state) throws Exception {
+        assertBusy(() -> {
+            String currentState = getDataFrameTaskState(transformId);
+            assertThat(state.value(), equalTo(currentState));
+        });
+    }
+
+    private void stopDataFrameTransform(String transformId, boolean force) throws Exception {
```

Comment:
This method could be moved to `DataFrameRestTestCase`

Also rename `startTransformRequest` -> `stopTransformRequest` and update or delete the comment.

Ditto  `startTransformResponse`